cgi-bin/DW/Controller/Editor.pm

```diff
@@ -0,0 +1,119 @@
+# DW::Controller::Editor
+#
+# Single-purpose routes for setting the editor userprops, which determine the
+# default formatting type when writing new entries/comments/etc in the web UI.
+#
+# Authors:
+#   Nick Fagerlund <nick.fagerlund@gmail.com>
+#
+# Copyright (c) 2020 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself.  For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+package DW::Controller::Editor;
+
+use strict;
+use LJ::JSON;
+use DW::Controller;
+use DW::Routing;
+use DW::Template;
+use DW::Formats;
+use Carp;
+
+DW::Routing->register_string( "/default_editor", \&default_editor_handler, app => 1 );
+
+DW::Routing->register_rpc(
+    "default_editor", \&default_editor_rpc_handler,
+    format  => 'json',
+    methods => { POST => 1 },
+);
+
+# Returns (1, "format_id") on success, (0, "error") on error.
+sub default_editor_helper {
+    my ( $remote, $type, $new_editor ) = @_;
+    my $userprop;
+
+    my $err = sub { return ( 0, $_[0] ); };
+
+    return $err->('nouser') unless $remote;
+
+    $new_editor = DW::Formats::validate($new_editor);
+
+    if ( $type eq 'comment' ) {
+        $userprop = 'comment_editor';
+    }
+    elsif ( $type eq 'entry' ) {
+        $userprop = 'entry_editor2';
+    }
+    else {
+        return $err->('unknowntype');
+    }
+
+    $remote->set_prop( $userprop, $new_editor );
+    return ( 1, $new_editor );
+}
+
+# zero-javascript version
+sub default_editor_handler {
+    my ( $ok, $rv ) = controller( form_auth => 1 );
+    return $rv unless $ok;
+
+    my $r      = $rv->{r};
+    my $POST   = $r->post_args;
```

Comment:
You will need to check `$r->did_post` to ensure that the caller actually did `POST` and return something valid if they didn't. That doesn't come for free in the standard handler. (It does in the RPC handler, when you specify that only `POST` methods are valid...)